ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoLocalAssembler-impl.h

```diff
@@ -0,0 +1,232 @@
+
+namespace ProcessLib
+{
+namespace TwoPhaseFlowWithPrho
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void TwoPhaseFlowWithPrhoLocalAssembler<
+    ShapeFunction, IntegrationMethod,
+    GlobalDim>::assemble(double const t, std::vector<double> const& local_x,
+                         std::vector<double>& local_M_data,
+                         std::vector<double>& local_K_data,
+                         std::vector<double>& local_b_data)
+{
+    auto const local_matrix_size = local_x.size();
+
+    assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+
+    auto local_M = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    auto local_b = MathLib::createZeroedVector<LocalVectorType>(
+        local_b_data, local_matrix_size);
+
+    NodalMatrixType mass_operator;
+    mass_operator.setZero(ShapeFunction::NPOINTS, ShapeFunction::NPOINTS);
+
+    auto Mgp =
+        local_M.template block<nonwet_pressure_size, nonwet_pressure_size>(
+            nonwet_pressure_matrix_index, nonwet_pressure_matrix_index);
+    auto Mgx = local_M.template block<nonwet_pressure_size, cap_pressure_size>(
+        nonwet_pressure_matrix_index, cap_pressure_matrix_index);
+
+    auto Mlp = local_M.template block<cap_pressure_size, nonwet_pressure_size>(
+        cap_pressure_matrix_index, nonwet_pressure_matrix_index);
+
+    auto Mlx = local_M.template block<cap_pressure_size, cap_pressure_size>(
+        cap_pressure_matrix_index, cap_pressure_matrix_index);
+
+    NodalMatrixType laplace_operator;
+    laplace_operator.setZero(ShapeFunction::NPOINTS, ShapeFunction::NPOINTS);
+
+    auto Kgp =
+        local_K.template block<nonwet_pressure_size, nonwet_pressure_size>(
+            nonwet_pressure_matrix_index, nonwet_pressure_matrix_index);
+
+    auto Kgx = local_K.template block<nonwet_pressure_size, cap_pressure_size>(
+        nonwet_pressure_matrix_index, cap_pressure_matrix_index);
+
+    auto Klp = local_K.template block<cap_pressure_size, nonwet_pressure_size>(
+        cap_pressure_matrix_index, nonwet_pressure_matrix_index);
+
+    auto Klx = local_K.template block<cap_pressure_size, cap_pressure_size>(
+        cap_pressure_matrix_index, cap_pressure_matrix_index);
+
+    auto Bg = local_b.template segment<nonwet_pressure_size>(
+        nonwet_pressure_matrix_index);
+
+    auto Bl =
+        local_b.template segment<cap_pressure_size>(cap_pressure_matrix_index);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    SpatialPosition pos;
+    pos.setElementID(_element.getID());
+    _process_data._material->setMaterialID(pos);
+
+    const Eigen::MatrixXd& perm = _process_data._material->getPermeability(
+        t, pos, _element.getDimension());
```

Comment:
I think it should be element's dimension, regarding this, @wenqing has made a comment in Liquid Flow process.